### DISCUSSION TITLE
.whereIdIn() ?

### DISCUSSION THREAD

**Discussion by bpuig** [CodeDredd/pinia-orm#427] [Q&A]
Hello, I'm migrating from vuex-orm and I'm stuck because `whereIdIn` seems to be missing .

As an alternative I've tried `whereIn`, but has issues.

```
// This does not work "Unresolved function or method whereIn() "
return teamUsersRepo
    .whereIn('id', userIds)
    .with('role')
    .get()
```

```
// This works, but does not respect order of the array in userIds
return teamUsersRepo
    .with('role')
    .whereIn('id', userIds)
    .get()
```

In vuex-orm `whereIdIn` worked, and also it kept the order given in the array on the results.

Will this function exist in the package?

**Comment by CodeDredd** [comment]
Try using

````js
return teamUsersRepo
    .query()
    .whereIn('id', userIds)
    .with('role')
    .get()
````

**Reply by bpuig** [reply]
Yesssssssss! :partying_face:

**Comment by bpuig** [comment]
I thought It worked, but now I see it does keep the order provided, **1st object are the `id` used in `whereIn`**, **2nd object is the result of the pinia-orm query**, the first order is the one provided by my API server:

![image](https://user-images.githubusercontent.com/22938625/200176486-3d6444b0-56b6-4a51-93c0-5d31871932c9.png)

Changed the order of the `id`s , but pinia stays the same.

![image](https://user-images.githubusercontent.com/22938625/200176558-32ab65bf-6324-4e96-8ee7-e22bd286d8a3.png)
Any idea where I should look?

I did not notice order was not working because I had an `orderBy` set, but then I saw that order is case-sensitive and that without `orderBy` it does not use the given ids order.

**Edit:** I see now that the order kept is the first call done to the API, then it stays frozen. Is there some kind of cache set by default? It's a bit weird that only respects the order first time and then when changing the ids object content it stays the same.

**Reply by CodeDredd** [reply]
Well there is a cache. But it is not on by default. I don't know yet if i understand the problem correctly. Can you maybe please make a reproduction 🙏  ?
